.github/workflows/next-build-nightly.yml

```diff
@@ -0,0 +1,155 @@
+        name: Check public api
+      - run: yarn test:self
+        name: Self tests for tools
+      - run: yarn check:circular
+        name: Circular dependency Check
+      - run: yarn doc
+        name: API Doc Check
+      - uses: ./.github/actions/check-license
+        name: License Check
+      - if: failure()
+        name: Slack Notify
+        uses: rtCamp/action-slack-notify@v2.3.2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+          SLACK_USERNAME: SDK Pipeline Bot
+          SLACK_TITLE: Build checks for v4 ${{ job.status }}
+          SLACK_MESSAGE: 'Test failed'
+          MSG_MINIMAL: event,actions url
+          SLACK_COLOR: ${{ job.status }}
+          SLACK_MSG_AUTHOR: ' '
+          SLACK_ICON: https://sap.github.io/cloud-sdk/img/logo.png
+  e2e-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: 'main'
+      - run: git fetch --depth=1
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: 'yarn'
+      - run: yarn install --frozen-lockfile --ignore-engines
+      - run: yarn test:e2e
+  canary-release-pre-check:
+    runs-on: ubuntu-latest
+    outputs:
+      skip: ${{ steps.date-check.outputs.skip-release }}
+    needs: [tests, checks, e2e-tests]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: 'main'
+      - run: git fetch --depth=1
+      - id: date-check
+        name: Check if latest commit is within 24 hrs
+        run: |
+          lastCommitDate=$(git --no-pager log -n 1 ${{ github.ref_name }} --pretty=format:"%at")
+          curDate=$(date +%s)
+          dateDiff=$(expr $curDate - $lastCommitDate)
+          echo $lastCommitDate, $curDate, $dateDiff
+
+          if [[ $dateDiff -gt 86400 ]]
+          then
+              echo 'No new commit found on ${{ github.ref }} within the last 24 hrs.'
+              echo "skip-release=true" >> $GITHUB_OUTPUT
+          else
+              echo "skip-release=false" >> $GITHUB_OUTPUT
+          fi
+  canary-release:
+    if: ${{ needs.canary-release-pre-check.outputs.skip == 'false' }}
+    runs-on: ubuntu-latest
+    needs: [canary-release-pre-check]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: 'main'
+      - run: git fetch --depth=1
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: 'yarn'
+          registry-url: 'https://registry.npmjs.org'
+      - run: yarn install --frozen-lockfile --ignore-engines
+      - name: Canary Release
+        run: |
+          date=`date +%Y%m%d%H%M%S`
+          rm -f .changeset/*.md
+          project_version=$(cat package.json | jq '.version' | tr -d '"')
+          echo The project version is: $project_version
+          [[ $project_version == 4* ]] && cp canary-release-changeset.md .changeset          
+          [[ $project_version == 3* ]] && cp major-release-changeset.md .changeset
```

Comment:
[req] I dont see a major-release-changeset.md in this PR

Reply:
I swapped the correct changeset files for the versions. But, I think in this workflow, we don't need the check for version 3 right?
Where do these changesets get published in `npm`? I wasn't easily able to find it now.

Reply:
> I swapped the correct changeset files for the versions. 

I didn't understand this.

> I think in this workflow, we don't need the check for version 3 right
The project version will always be 3 here, as thats the version in package.json. Hence we use the major release changeset so a version 4 next is released. We probably dont need a check for version 4. But I would test this first, else leave it here.

Reply:
Discussed in call, left it as is, as this job is a copy paste of the `build` workflow in main.